### PR TITLE
Refactor DOW doc template

### PR DESCRIPTION
**Pull request by juliusfitzhugh-ccpo**
Refactor DOW docx template along with adjust logic as needed to fit the schema 
and how the UI is saving data. The main change was around the `xaasOfferings` 
property in the payload which was refactored because specific tables are used
to capture certain information related to the XaaS offerings (e.g., Selected
Service Offering is only used for 7 of the XaaS services not all). 

```typescript
// old schema
xaasOffering: [
  serviceOffering: {},
  instanceConfigurations:[],
]

// new schema
xaasOffering: {
  computeInstances: [];
  databaseInstances: [];
  storageInstances: [];
  generalInstances: [];
  selectedServiceInstances: [];
}
```

### Update `dow-document` function
Update function with any new or refactored logic to use when generating
the DOW document. 

### Add/update util functions
Util functions specifically for the `Dow` are placed in the `utils/dow.ts`
file to help format data in the template easier.

### Update models and sample data
Additional properties were added if missing or updated to account for how the data
is sent to the HOTH API. The sample data was updated  with these changes as well.

Currently, the main validation for the payload sent to the HOTH API
has the top level properties needed to fill in the template see `descriptionOfWork` 
in `models/document-generation.ts`. 

This is not to be confused with models in `models/document-generation/description-of-work.ts` 
which is an initial start of splitting out models related to the DOW and is in a transitioning phase.
Some of the models are not currently being used, but have more nested properties, and are
not being used currently to reduce friction with generating the dow template.

Ticket: AT-8380

### CODE REVIEW

File: document-generation/utils/helpers.ts [hunk 1]
```diff
@@ -28,8 +28,14 @@ export const formatDuration = (periods: IPeriod[]): string => {
     ? base.map((period) => `Base: ${period.periodUnitCount} ${period.periodUnit}`).join("")
     : "";
   const hasOptions = options.length > 0;
+  // The UI sets the `optionOrder` for the Base period as `1` and the Option orders follow
+  // consecutively (e.g, OP1 = 2, OP2 = 3, etc). Given the UI does not allow for an Option
+  // period to be skipped when creating a Period of Perforamnce (PoP), it is assumed there
+  // will be no gaps in the PoP option periods (e.g., B, OP1, OP3). As such, the display value
+  // is generated based on an array of the option periods index rather than the `optionOrder`
+  // previously used.
```
Comment (jasonburkert-ccpo) [inline document-generation/utils/helpers.ts:36]: Love the background

File: document-generation/utils/dow.ts [hunk 2]
```diff
@@ -0,0 +1,1113 @@
+import {
```
Comment (jasonburkert-ccpo) [inline document-generation/utils/dow.ts:1]: Yikes. Intense.

**Review by jasonburkert-ccpo — APPROVED** [review]
Very nice document output.  Holy-bananas, that template file!
Nice testing effort.

**Comment by jasonburkert-ccpo** [comment]
@jeffsegal-ccpo Can we override the coverage threshold somehow and get this merged?

File: document-generation/utils/dow.ts (hunk 2 above)
Comment (juliusfitzhugh-ccpo) [reply document-generation/utils/dow.ts:1]: @jasonburkert-ccpo this file will definitely need some refactoring along with the addition of tests. We should create a tech debt ticket to be capture that.

**Pull request merged**